Gems/ROS2/Code/Source/RobotImporter/Utils/RobotImporterUtils.cpp

```diff
@@ -324,4 +325,55 @@ namespace ROS2
         return resolvedPath.String();
     }
 
+    namespace Utils::SDFormat
+    {
+        AZStd::string GetPluginFilename(const sdf::Plugin& plugin)
+        {
+            const std::filesystem::path path = plugin.Filename();
+            return AZStd::string(path.filename().u8string().c_str(), path.filename().u8string().size());
+        }
```

Comment:
Instead of adding the `<filesystem>` include an using std::filesystem, this code can go directly to an AZ::IO::Path.
I also think that would be a better return value than AZStd::string, since that enforces specifically what kind of value is the result of the function returns
```suggestion
        AZ::IO::Path GetPluginFilename(const sdf::Plugin& plugin)
        {
            const std::string& path = plugin.Filename();
            return AZStd::string(path.c_str(), path.size());
        }
```

Reply:
The goal of this method is to strip the possible path and keep the filename only. As pointed out by Mike, the filename tag could include the folder or not [source](http://sdformat.org/spec?ver=1.10&elem=model#plugin_filename). The matching is based on the plugin names solely. 

I removed the `<filesystem>` include and used `AZ::IO::Path` instead, which simplified the code. Thanks for reminding me about not using `std` when not needed.